Dashboard

CassandraTolton opened the pull request:
# Summary
updating the skeleton code and further laying out what needs to be done. nothing complex.

# Description
this is bare bones skeleton code. until we finalize the way we are linking pages that feature will be left unfinished. what is on the page are div containers and buttons for the compare students and alerts.

# Documentation
variables used for the page will be the following:
    Variables: Class_Name, Section_Name, Display_Name
    Buttons: Menu, Student Comparison, Alerts, n boxes with Class_Names and Section_Names
    Links: Student Comparison Tool, Alerts, Each box
    Header takes Display_Name

documentation on how the page will function is found in the discord - additional documentation will be made at a later time. the wireframe is from Jacobs wireframe in the discord. 

# Unit Tests
no unit tests were made are there is no functionality currently to test

# PR Reviews
front end should review and see if any changes or additions need to be made to the structure of the code. 

for what needs to be done:
1. linking of the pages to the compare students and alerts pages
2. creating and linking to the sections and their assignments
3. creation of the banner and header for the welcome text
4. further stylization and completion 

![image_2025-03-24_201331187](https://github.com/user-attachments/assets/997355b9-ca34-41ae-92d9-71a04606fcdb)

once linking and querey are added, tests will be made to ensure that the query is correct and that the changing of the pages is successful.

EvanShumaker committed:
Merge pull request #19 from UNLV-CS472-672/dev

Created Templates for PRs and Issues.

CassandraTolton committed:
Dashboard

# Summary
updating the skeleton code and further laying out what needs to be done. nothing complex.

# Description
this is bare bones skeleton code. until we finalize the way we are linking pages that feature will be left unfinished. what is on the page are div containers and buttons for the compare students and alerts.

# Documentation
variables used for the page will be the following:
    Variables: Class_Name, Section_Name, Display_Name
    Buttons: Menu, Student Comparison, Alerts, n boxes with Class_Names and Section_Names
    Links: Student Comparison Tool, Alerts, Each box
    Header takes Display_Name

documentation on how the page will function is found in the discord - additional documentation will be made at a later time. the wireframe is from Jacobs wireframe in the discord.

# Unit Tests
no unit tests were made are there is no functionality currently to test

# PR Reviews
front end should review and see if any changes or additions need to be made to the structure of the code.

for what needs to be done:
1. linking of the pages to the compare students and alerts pages
2. creating and linking to the sections and their assignments
3. creation of the banner and header for the welcome text
4. further stylization and completion

![image_2025-03-24_201331187](https://github.com/user-attachments/assets/997355b9-ca34-41ae-92d9-71a04606fcdb)

once linking and querey are added, tests will be made to ensure that the query is correct and that the changing of the pages is successful.

Bakera21 commented:
Looks like a good start, once the pages are ready we can start linking up everything. I would say, if you can, prep the buttons to link to the other pages, even though there aren't any pages to link to yet. Great documentation.

jkasbohm commented:
This looks good. If you want tips on how to do the routing between the pages for the compare students and alerts buttons, you can look at the files in either mine or Steven's branches (features/login & features/authentication). That should help with getting the routing setup for your next PR.

CassandraTolton committed:
Updated Dashboard

updated dashboard

added code from stevens dashboard page

referenced jacobs sign in page

added classnames to each div for prep. of styling

added a function that adds the buttons (this will be used to create the sections)

added links to the two buttons; student comparisons and alerts

added a div for the header and a button for the menu

made additional formatting changes for easier readability

added more comments

Co-Authored-By: Steven Fojas <84162024+stevenfojas670@users.noreply.github.com>

CassandraTolton committed:
Merge branch 'UNLV-CS472-672:master' into Dashboard

CassandraTolton reviewed (commented):
added linking by looking at steven and jacobs files, added more buttons, a div for the header and some of stevens code that was present in his pr.
